woocommerce/woocommerce-ios
# [Mobile Payments] Add filled red badge to the HubMenu buttons

Closes: #7359 


### Description

With this PR we prepare the badge to be rendered with filled red color and shown in the Payments Hub Menu button. To do that we added a new enum `HubMenuBadgeType` to represent the two different types of badges that we will support. The logic for displaying it is still not in place, so you won't see anything out of the box. To show it you can change the lines [here](https://github.com/woocommerce/woocommerce-ios/blob/ee1dab6fd6300172059a4286eb62ca0e04e631cd/WooCommerce/Classes/ViewRelated/Hub%20Menu/HubMenuViewModel.swift#L201) to:

```
var badge: HubMenuBadgeType {
            switch self {
            case .payments:
                return .newFeature
            default:
                return .number(number: 0)
            }
        }
```

to show the .`newFeature` type badge.

### Testing instructions


1. Add the replacement mentioned above
2. Go to Menu
3. You should see the badge on the Payments button:

### Screenshots


<img src="https://user-images.githubusercontent.com/1864060/182189328-229ec287-d3ad-4d04-af11-2d524038a37e.png" width="375">

---
